internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go

```diff
@@ -438,7 +460,11 @@ func (h *Handler) getShare(w http.ResponseWriter, r *http.Request, shareID strin
 		response.WriteOCSError(w, r, response.MetaServerError.StatusCode, "error mapping share data", err)
 	}
 	h.mapUserIds(ctx, client, share)
-
+	if receivedShare != nil && receivedShare.State == collaboration.ShareState_SHARE_STATE_ACCEPTED {
+		// Needed because received shares can be jailed in a folder in the users home
+		share.FileTarget = path.Join(h.sharePrefix, path.Base(info.Path))
```

Comment:
This breaks global urls for accessing shares as you're forcing a basePath for `FileTarget` and `Path`. 
It needs https://github.com/cs3org/reva/pull/1739 to be merged first and apply the same logic if I didn't miss anything.

Reply:
@C0rby I just merged https://github.com/cs3org/reva/commit/afa06f7d83d5751b94e37e7bc3e1c5ef9580e06a, can you update this part with the same logic. If you're in the mood, it would make sense to refactor that `switch` logic into a function